### ISSUE TITLE
[Accessibility] Button triggering modal missing aria-haspopup

### ISSUE THREAD

**Issue by amberhinds** [woocommerce/woocommerce-gateway-amazon-pay#330]
## Audit Information
This issue came from an audit WooCommerce had Equalize Digital complete in December 2025. I'm moving the issue out of Linear (WooA11Y-427) to this repo.

- EQD Audit ID: WooExt-520
- Component/Page: Checkout
- Affected Element: Select button
- URL where issue was discovered: https://apay-woo.mystagingwebsite.com/checkout/

## WCAG Compliance

- Success Criterion: 4.1.2 Name, Role, Value
- WCAG Version: 2
- Level: A
- Severity: Medium

## Issue Description

When a button triggers a modal or dialog to open, this should be communicated to screen reader users to set expectations for the action. The aria-haspopup attribute serves this purpose, but it is missing in this case.

### Impacted Populations

Blind and low vision people, Screen reader users (blind/low vision and neurodivergent people)

## Recommended Fix

Add aria-haspopup="dialog" to the button so screen reader users are informed of the action when the button is activated. Additionally, include the aria-controls attribute to link the button to the ID of the modal being opened.

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup  and https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls

## Code Example

`<a href="#" class="wc-apa-widget-change amazonpay-merchant-shadow-root-parent-element-for-executing-modal-script" id="amazon_change_payment_method" data-wc_apa_change_bind="changePayment">Select</a>`

## Screenshots

https://drive.google.com/file/d/1RH4M5GsHh8n4g9qYM4denLVdd1_BermR/view?usp=drivesdk
